app/models/user.rb

```diff
@@ -57,6 +57,21 @@ class User < ApplicationRecord
   validates :full_name, :gender, presence: true
   validates :gender, inclusion: { in: genders }
 
+  def self.from_social_provider(provider, user_params)
```

Comment:
I think it's better to move this to a UserService or something like that.
But it's okay, you don't need to change it.

Reply:
I literally copied most of facebook login from [here](https://github.com/rootstrap/seatmate_api)